src/esp/sensor/FisheyeSensor.cpp

```diff
@@ -147,12 +147,41 @@ bool FisheyeSensor::drawObservation(sim::Simulator& sim) {
     }
   }
 
-  esp::gfx::RenderCamera::Flags flags;
+  esp::gfx::RenderCamera::Flags flags = {gfx::RenderCamera::Flag::ClearColor |
+                                         gfx::RenderCamera::Flag::ClearDepth};
   if (sim.isFrustumCullingEnabled()) {
     flags |= gfx::RenderCamera::Flag::FrustumCulling;
   }
+
   // generate the cubemap texture
-  cubeMap_->renderToTexture(*cubeMapCamera_, sim.getActiveSceneGraph(), flags);
+  if (fisheyeSensorSpec_->sensorType == SensorType::Semantic) {
+    bool twoSceneGraphs =
+        (&sim.getActiveSemanticSceneGraph() != &sim.getActiveSceneGraph());
+
+    if (twoSceneGraphs) {
+      moveSemanticSensorToSemanticSceneGraph(sim);
+      cubeMap_->renderToTexture(*cubeMapCamera_,
+                                sim.getActiveSemanticSceneGraph(), flags);
+      moveSemanticSensorBackToRegularSceneGraph(sim);
```

Comment:
@Skylion007 : I reduced the code duplication by extracting 2 helper functions applied here.

The challenge is that the code looks like a "sandwich". The "bread" is the same but the "food" in between varies.